why-django.qmd

```diff
@@ -25,12 +23,11 @@ State the context and some background on the issue, then write a
 statement in the form of a question for the problem.
 :::
 
-For a software project to succeed it is important to pick the right
-programming framework to work in from the start. A framework should
-provide consistency, code reuse, and easy of maintenance for a group of
-developers. It should provide best practice, and allow for code to be
-shared between different parts of the applications. The question
-therefore becomes:
+Web applications all have a host of common functionalities (e.g. HTTP request handling, URL routing or data source integration), which take significant time and effort to implement from scratch and maintain for each new project. Good frameworks provide mature, well-maintained components for these features out of the box as well as best-practice patterns and guidelines for structuring code consistently. They also promote code reuse and modular design both within and between applications.
```

Comment:
```suggestion
Web applications all have a host of common functionalities (e.g. HTTP request handling, URL routing or data source integration). These can take significant time and effort to implement from scratch and maintain for each new project. Good frameworks provide mature, well-maintained components for these features out of the box as well as best-practice patterns and guidelines for structuring code consistently. They also promote code reuse and modular design both within and between applications.
```
- Re Oxford comma: unless there's a style rule for always using it, I wouldn't use it here because the list is not super complex
- Re "good": to me it feels like the entire section is about explaining what a good framework is, so I would actually prefer to keep it vague at the beginning and clarify it as we go along

Reply:
In my strong opinion, the Oxford comma should always be used if a list of three or more can't be simplified. And there's strong reasons for using it always e.g. https://thewritelife.com/is-the-oxford-comma-necessary/ we don't follow a specific writing style guide (might be good to consider one). Regardless of the guide though, I will always say to use the serial aka Oxford comma.